Source/AGXUnrealBarrier/Private/Terrain/ShovelBarrier.cpp

```diff
@@ -157,152 +157,152 @@ void FShovelBarrier::SetMinimumSubmergedContactLengthFraction(
+	NativeRef->Native->getSettings()->getAdvancedSettings()->setMinimumSubmergedContactLengthFraction(
 		MinimumSubmergedContactLengthFraction);
 }
 
 double FShovelBarrier::GetMinimumSubmergedContactLengthFraction() const
 {
 	check(HasNative());
-	return NativeRef->Native->getMinimumSubmergedContactLengthFraction();
+	return NativeRef->Native->getSettings()->getAdvancedSettings()->getMinimumSubmergedContactLengthFraction();
 }
 
 void FShovelBarrier::SetVerticalBladeSoilMergeDistance(double VerticalBladeSoilMergeDistance)
 {
 	check(HasNative());
 	const agx::Real DistanceAGX = ConvertDistanceToAGX(VerticalBladeSoilMergeDistance);
-	NativeRef->Native->setVerticalBladeSoilMergeDistance(DistanceAGX);
+	NativeRef->Native->getSettings()->setVerticalBladeSoilMergeDistance(DistanceAGX);
 }
 
 double FShovelBarrier::GetVerticalBladeSoilMergeDistance() const
 {
 	check(HasNative());
-	const agx::Real DistanceAGX = NativeRef->Native->getVerticalBladeSoilMergeDistance();
+	const agx::Real DistanceAGX = NativeRef->Native->getSettings()->getVerticalBladeSoilMergeDistance();
 	const double DistanceUnreal = ConvertDistanceToUnreal<double>(DistanceAGX);
 	return DistanceUnreal;
 }
 
 void FShovelBarrier::SetSecondarySeparationDeadloadLimit(double SecondarySeparationDeadloadLimit)
 {
 	check(HasNative());
-	NativeRef->Native->setSecondarySeparationDeadloadLimit(SecondarySeparationDeadloadLimit);
+	NativeRef->Native->getSettings()->getAdvancedSettings()->setSecondarySeparationDeadloadLimit(SecondarySeparationDeadloadLimit);
 }
 
 double FShovelBarrier::GetSecondarySeparationDeadloadLimit() const
 {
 	check(HasNative());
-	return NativeRef->Native->getSecondarySeparationDeadloadLimit();
+	return NativeRef->Native->getSettings()->getAdvancedSettings()->getSecondarySeparationDeadloadLimit();
 }
 
 void FShovelBarrier::SetPenetrationDepthThreshold(double PenetrationDepthThreshold)
 {
 	check(HasNative());
 	const agx::Real DepthAGX = ConvertDistanceToAGX(PenetrationDepthThreshold);
-	NativeRef->Native->setPenetrationDepthThreshold(DepthAGX);
+	NativeRef->Native->getSettings()->setPenetrationDepthThreshold(DepthAGX);
 }
 
 double FShovelBarrier::GetPenetrationDepthThreshold() const
 {
 	check(HasNative());
-	const agx::Real DepthAGX = NativeRef->Native->getPenetrationDepthThreshold();
+	const agx::Real DepthAGX = NativeRef->Native->getSettings()->getPenetrationDepthThreshold();
 	const double DepthUnreal = ConvertDistanceToUnreal<double>(DepthAGX);
 	return DepthUnreal;
 }
 
 void FShovelBarrier::SetPenetrationForceScaling(double PenetrationForceScaling)
 {
 	check(HasNative());
-	NativeRef->Native->setPenetrationForceScaling(PenetrationForceScaling);
+	NativeRef->Native->getSettings()->setPenetrationForceScaling(PenetrationForceScaling);
 }
 
 double FShovelBarrier::GetPenetrationForceScaling() const
 {
 	check(HasNative());
-	return NativeRef->Native->getPenetrationForceScaling();
+	return NativeRef->Native->getSettings()->getPenetrationForceScaling();
 }
```

Comment:
Are these changes compatible with older AGX versions? If not, this PR will need to wait until we target AGX 2.40 if we don't want to add version checks.